### PR TITLE
CDAT Migration: Refactor `meridional_mean_2d` set

### LINKED ISSUE #657: CDAT Migration Phase 2: Refactor `meridional_mean_2d` set

### PR DESCRIPTION
**Pull request by tomvothecoder**
- Close #657 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] New and existing unit tests pass with my changes (locally and CI/CD build)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: e3sm_diags/plot/utils.py [hunk 1]
```diff
@@ -580,7 +580,7 @@ def _add_min_mean_max_text(
     fontdict = {"fontsize": fontsize}
 
     if left_text_pos is None:
-        left_text_pos = (0.6335, 0.2107)
+        left_text_pos = (0.6635, 0.2107)
```
Comment (tomvothecoder) [inline e3sm_diags/plot/utils.py:583]: Fix incorrect float for left text position adjustment

**Review by tomvothecoder — COMMENTED** [review]
Hi @chengzhuzhang, this PR is ready for review. I added comments to areas of interest in the file diffs. All of the `.nc` files are within relative tolerance.

Here are the links to the viewers:
- `main`: https://portal.nersc.gov/project/e3sm/cdat-migration-fy24/main/viewer/meridional_mean_2d/
- dev branch: https://portal.nersc.gov/project/e3sm/cdat-migration-fy24/657-meridional-mean-2d/viewer/meridional_mean_2d/

I did an eye-check and all the plots look the same. The only difference I saw was that some of the "Max" metrics value for the second sub-plot differ, which might be due to using xESMF instead of Regrid2.

File: e3sm_diags/driver/meridional_mean_2d_driver.py [hunk 2]
```diff
@@ -20,185 +26,225 @@
...
+    ds_r_plevs_avg = _add_placeholder_y_axis(ds_ref, ds_r_plevs_avg, var_key)
+
+    ds_t_plevs_rg_avg, ds_r_plevs_rg_avg = align_grids_to_lower_res(
+        ds_t_plevs_avg,
+        ds_r_plevs_avg,
+        var_key,
+        tool="xesmf",
+        method="conservative_normed",
+        axis_to_compare="X",
+    )
```
[the remark this answers is not in the document]
Comment (chengzhuzhang) [reply e3sm_diags/driver/meridional_mean_2d_driver.py:120]: I recall that the default regridding back end for the `main` is `esmf`. And the results look good!

**Review by chengzhuzhang — APPROVED** [review]
The PR looks good! Thank you.

File: e3sm_diags/driver/meridional_mean_2d_driver.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomvothecoder) [reply e3sm_diags/driver/meridional_mean_2d_driver.py:120]: `"regrid2"` is hard-coded as the regrid method in `meridional_mean_2d_driver.py`. It was probably supposed to be replaced with `esmf` at some poit.

https://github.com/E3SM-Project/e3sm_diags/blob/43f5f74057929c4bedddfab58be63f838ee36abe/e3sm_diags/driver/meridional_mean_2d_driver.py#L150

https://github.com/E3SM-Project/e3sm_diags/blob/43f5f74057929c4bedddfab58be63f838ee36abe/e3sm_diags/driver/meridional_mean_2d_driver.py#L173

[the remark this answers is not in the document]
Comment (tomvothecoder) [reply e3sm_diags/driver/meridional_mean_2d_driver.py:120]: I verified that this was the only set driver that had "regrid2" hard-coded

[the remark this answers is not in the document]
Comment (chengzhuzhang) [reply e3sm_diags/driver/meridional_mean_2d_driver.py:120]: Thank you, @tomvothecoder. I can't recall why we made the choice for using `regrid2` when first implemented it. It is good to replace with `esmf` at this point.

**Pull request merged**